internal/dns/dns.go

```diff
@@ -216,7 +213,7 @@ func (dns *DNS) getMetricURLs() (map[string]string, error) {
 	urls := make(map[string]string)
 	for _, pod := range pods.Items {
 		if pod.Status.PodIP != "" {
-			urls[pod.Name] = fmt.Sprintf("http://%s:%s/metrics", pod.Status.PodIP, port)
+			urls[pod.Name] = fmt.Sprintf("http://%s:%d/metrics", pod.Status.PodIP, port)
```

Comment:
I'm not really suggesting a change here, but I've had it pointed out to me before, that it is better to build a URL using a net/url.URL value rather than Sprintf. But, I don't think its really necessary here, because I don't think there is any chance your data in pod could contain invalid IP address data, and I don't know that it would even matter if a bad URL got generated here.

Reply:
definitely worth looking into, thanks!